### DISCUSSION TITLE
is it possible an old-fashioned batch version will be developed

### DISCUSSION THREAD

**Discussion by tiger-tiger** [pyg-team/pytorch_geometric#4599] [General]
Is it possible there is an option, we can just use [B, F, N] for nodes, [B, F, E, E] for adj matrix as inputs, pyG automatically converts them to some underlying data format.

meanwhile we generate batched graphs with nodes [B, F,N] & adj [B, F, E, E] of same size, so loss can be easily calculated for labeled graphs.

**Comment by rusty1s** [comment]
We might be able to integrate something like that once [`NestedTensor`](https://github.com/pytorch/nestedtensor) is in a more stable state. No promise though :)
